src/engine/Map.cxx

```diff
@@ -382,13 +382,50 @@ Point Map::findNodeInMap(const SDL_Point &screenCoordinates) const
 {
   Point foundCoordinates{-1, -1, 0, 0};
 
-  for (auto &it : mapNodes)
+  // calculate clicked column (x coordinate) without height taken into account.
+  const Point calculatedIsoCoords = calculateIsoCoordinates(screenCoordinates);
+  int isoX = calculatedIsoCoords.x;
+  int isoY = calculatedIsoCoords.y;
+
+  // adjust caluclated valies that are outside of the map (which is legit, but they need to get pushed down)
+  if (isoY >= Settings::instance().mapSize)
+  {
+    int diff = isoY - Settings::instance().mapSize + 1; // +1 here so we are below the max value
+    isoY -= diff;
+    isoX += diff;
+  }
+  if (isoX >= Settings::instance().mapSize)
   {
-    if (isClickWithinTile(screenCoordinates, it->getCoordinates().x, it->getCoordinates().y) &&
-        (foundCoordinates.z <= it->getCoordinates().z))
+    int diff = isoY - Settings::instance().mapSize + 1;
```

Comment:
i'm not sure, but could this be wrong ? 
I think it should be isoX - ...  not isoY

Reply:
actually this isn't needed for the x coordinate. but i found a few other bugs and fixed them. now it should be finally correct